storage/cache/subtree_cache.go

```diff
@@ -169,8 +169,26 @@ func (s *SubtreeCache) preload(ids []storage.NodeID, getSubtrees GetSubtreesFunc
 	if err != nil {
 		return err
 	}
+
+	ch := make(chan *storagepb.SubtreeProto, len(want))
+	wg := &sync.WaitGroup{}
+
 	for _, t := range subtrees {
-		s.populate(t)
+		t := t
```

Comment:
Not sure about the CPU usage implications of running multiple hstar2 hashes in parallel. Might need to bound this.

Reply:
Good call - done.